app/client/src/layoutSystems/anvil/integrations/sagas/anvilWidgetAdditionSagas/helpers.ts

```diff
@@ -0,0 +1,249 @@
+import type {
+  FlattenedWidgetProps,
+  WidgetBlueprint,
+} from "WidgetProvider/constants";
+import { getNextEntityName } from "utils/AppsmithUtils";
+import type { DataTree } from "entities/DataTree/dataTreeTypes";
+import { getDataTree } from "selectors/dataTreeSelectors";
+import WidgetFactory from "WidgetProvider/factory";
+import { executeWidgetBlueprintOperations } from "sagas/WidgetBlueprintSagas";
+import { call, put, select } from "redux-saga/effects";
+import type { CanvasWidgetsReduxState } from "reducers/entityReducers/canvasWidgetsReducer";
+import AppsmithConsole from "utils/AppsmithConsole";
+import { ENTITY_TYPE } from "@appsmith/entities/AppsmithConsole/utils";
+import { WidgetReduxActionTypes } from "@appsmith/constants/ReduxActionConstants";
+
+/**
+ * In Anvil, we maintain some properties set by users on widgets.
+ * When a similar or same widget is added to the Canvas, we retrieve these
+ * properties and apply them.
+ * This is to assist users by automatically apply settings based on context.
+
+ * Retrieves the values from session storage for the widget properties
+ * for hydration of the widget when we create widget on drop
+ */
+export function getWidgetSessionValues(
+  type: string,
+  parent: FlattenedWidgetProps,
+) {
+  // For WDS_INLINE_BUTTONS_WIDGET, we want to hydation only to work when we add more items to the inline button group.
+  // So we don't want to hydrate the values when we drop the widget on the canvas.
+  if (["WDS_INLINE_BUTTONS_WIDGET"].includes(type)) return;
+
+  let widgetType = type;
+  const configMap = WidgetFactory.widgetConfigMap.get(type);
+
+  const widgetSessionValues: any = {};
+
+  // in case we are dropping WDS_ICON_BUTTON_WIDGET, we want to reuse the values of BUTTON_WIDGET
+  if (type === "WDS_ICON_BUTTON_WIDGET") {
+    widgetType = "WDS_BUTTON_WIDGET";
+  }
+
+  for (const key in configMap) {
+    if (configMap[key] != undefined) {
```

Comment:
nit: 
```suggestion
    if (configMap[key] != undefined) {
    if (configMap[key] == undefined) continue;
```

This will remove one nested level which increase readability a little.